app/controllers/users/registrations_controller.rb

```diff
@@ -16,15 +16,28 @@ def destroy
   end
 
   def update
-    if resource.update_with_password(resource_params)
-      render json: {
-        status: { code: 200, message: 'User information updated successfully.',
-                  data: UserSerializer.new(resource).serializable_hash[:data][:attributes] }
-      }, status: :ok
+    if resource.provider.present?
+      if resource.update(resource_params)
+        render json: {
+          status: { code: 200, message: 'User information updated successfully.',
+                    data: UserSerializer.new(resource).serializable_hash[:data][:attributes] }
+        }, status: :ok
+      else
+        render json: {
+          status: { message: "User information couldn't be updated. #{resource.errors.full_messages.to_sentence}" }
+        }, status: :unprocessable_entity
+      end
     else
-      render json: {
-        status: { message: "User information couldn't be updated. #{resource.errors.full_messages.to_sentence}" }
-      }, status: :unprocessable_entity
+      if resource.update_with_password(resource_params)
+        render json: {
+          status: { code: 200, message: 'User information updated successfully.',
+                    data: UserSerializer.new(resource).serializable_hash[:data][:attributes] }
+        }, status: :ok
+      else
+        render json: {
+          status: { message: "User information couldn't be updated. #{resource.errors.full_messages.to_sentence}" }
+        }, status: :unprocessable_entity
```

Comment:
try to reduce nesting if else conditions.

Reply:
reduced the nesting if-else.